modules/packages/LinearAlgebra.chpl

```diff
@@ -2500,6 +2517,410 @@ proc cosm(A: []) throws {
   }
 }
 
+// Gets sums along specified axes. Method for Rectangular matrix.
+proc absSum(A: [?D], axis=0) throws where !isCSDom(D) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  if axis == 0 then
+    // axis=0 sums along column
+    return forall j in D.dim(1-axis) do (+ reduce abs(A[D.dim(axis), j]));
+  else
+    // axis=1 sums along row
+    return forall i in D.dim(1-axis) do (+ reduce abs(A[i, D.dim(axis)]));
+}
+
+// Gets sums along specified axes. Method for CSR matrix.
+private proc absSum(A: [?ADom], axis=0) throws where isCSDom(ADom) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[j] += abs(A[i,j]);
+    }
+    return b;
+  }
+  else {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[i] += abs(A[i,j]);
+    }
+    return b;
+  }
+}
+
+// Gets max along specified axes. Method for CSR matrix.
+private proc maxAlongAxis(A: [?ADom], axis=0) throws where isCSDom(ADom) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
```

Comment:
This goes into the overarching topic of "what should the return array X look like for a given function Y with the argument Z?" question.

Here, the user could pass a non-zero based `A`, but here you're returning the maximums along a dimension in a 0-based vector. I think your `b`'s should be `var b: [ADom.dim(1-axis)] A.eltType`

Note that once you make that change your if/else blocks will be even more similar. Then you can change the forall to look like

```
forall idx in ADom with (max reduce b) {
  const bIdx = idx[1-axis];
  b[bIdx] = max(b[bIdx], A[i,j]);
}
```

which can allow you to drop the condition altogether.